src/streamlink/plugins/twitch.py

```diff
@@ -216,27 +216,28 @@ def call_subdomain(self, subdomain, path, format="json", schema=None, **extra_pa
         self.subdomain = subdomain_buffer
         return response
 
-    def access_token(self, endpoint, asset, **params):
-        return self.call("/api/{0}/{1}/access_token".format(endpoint, asset), **params)
+    # Public API calls
+
+    def user(self, **params):
+        return self.call("/kraken/user", **params)
+
+    def videos(self, video_id, **params):
+        return self.call("/kraken/videos/{0}".format(video_id), **params)
 
     def channel_info(self, channel, **params):
         return self.call("/kraken/channels/{0}".format(channel), **params)
 
-    def channel_subscription(self, channel, **params):
-        return self.call("/api/channels/{0}/subscription".format(channel), **params)
+    def channel_search(self, channel, **params):
+        return self.call("/kraken/search/channels", query=channel, **params)
```

Comment:
Do we still need the channel_search method now that we are using the other API call?